src/main/fc/rc.c

```diff
@@ -421,21 +416,23 @@ static FAST_CODE void processRcSmoothingFilter(void)
         }
     }
 
-    // each pid loop, apply the last received channel value to the filter, if initialised - thanks @klutvott
-    for (int i = 0; i < PRIMARY_CHANNEL_COUNT; i++) {
-        float *dst = i == THROTTLE ? &rcCommand[i] : &setpointRate[i];
-        *dst = pt3FilterApply(&rcSmoothingData.filterSetpoint[i], rxDataToSmooth[i]);
-    }
+    if (rxConfig()->rc_smoothing) {
```

Comment:
Another spot where the logic can be inverted and the rest pulled out of the if-statement, to make the data flow a bit easier to follow
```suggestion
    if (!rxConfig()->rc_smoothing) return;

```